File: examples/swift/Hello.swift
// Swift branch of cast's examples corpus. Pointed at by the
// `swift_examples` concept via cast::io::continues_in!.

func greet(name: String) -> String {
    return "Hello, \(name)!"
}

func main() {
    print(greet(name: "World"))
}

protocol Greeter {
    var language: String { get }
    var salutation: String { get }
}

File: examples/typescript/hello.ts
// TypeScript branch of cast's examples corpus. Pointed at by the
// `typescript_examples` concept via cast::io::continues_in!.

function greet(name: string): string {
    return `Hello, ${name}!`;
}

function main(): void {
    console.log(greet("World"));
}

interface Greeter {
    language: string;
    salutation: string;
}

const enGreeter: Greeter = { language: "en", salutation: "Hello" };
